### PR TITLE
[ISSUE#6116] Refactor ClientManager and make register client and schedule checkTask is atomic.

### LINKED ISSUE #6116: Unable to check client beats when register same ip and port in old clients with high concurrent
**Describe the bug**
When users register different services which has same ip and port in old clients and 2.0.x servers with high concurrent, The server may has some problem and cause the instance can't check the beats correctly.

The reason is in `com.alibaba.nacos.naming.core.InstanceOperatorClientImpl#createIpPortClientIfAbsent`. 

containClient, clientConnected and instance IpPortBasedClient is not atomic operation and make the beatCheckTask which async to do is mismatch the right client.

situation like under:
```

TA  containClient --> false --> instance ClientA -----> schedule BeatCheckTaskA -------------------------> clientConnected

TB  containClient --> false -------> instance ClientB ------> schedule BeatCheckTaskB ------> clientConnected

```  

So that clientManager cache the ClientA but schedule BeatCheckTaskB.

**Expected behavior**
expired instance can be removed.

**Acutally behavior**
expired instance can't be removed and set unhealthy.

**How to Reproduce**
1. start any mode server 2.0.x.
2. register same ip and port in old clients with high concurrent.
3. shutdown all instance.
4. check whether instance will be removed. 

**Desktop (please complete the following information):**
 - Version nacos-server 2.0.x + nacos-client 1.X
 - Module naming

**Additional context**

Fix way:

make the instance Client(schedule BeatCheckTask) and register client to manager is atomic operation will be ok.

### PR DESCRIPTION
**Pull request by KomachiSion**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

Fixes #6116 .

## Brief changelog

- Refactor ClientManager and make register client and schedule checkTask is atomic.

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java [hunk 1]
```diff
@@ -170,7 +171,10 @@ public Response onApply(WriteRequest request) {
     
     private void onInstanceRegister(Service service, Instance instance, String clientId) {
         Service singleton = ServiceManager.getInstance().getSingleton(service);
-        Client client = clientManager.computeIfAbsent(clientId, () -> new IpPortBasedClient(clientId, false));
+        if (!clientManager.contains(clientId)) {
```
Comment (pixystone) [inline naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:174]: This `contains` is unnecessary.

File: naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java [hunk 2]
```diff
@@ -170,7 +171,10 @@ public Response onApply(WriteRequest request) {
...
+            clientManager.clientConnected(clientId, new ClientAttributes());
```
Comment (pixystone) [inline naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:175]: `ephemeral` is removed but never set at another place.

**Review by pixystone — CHANGES_REQUESTED** [review]

File: naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java [hunk 3]
```diff
@@ -289,6 +293,7 @@ protected void loadSnapshot(byte[] snapshotBytes) {
             ConcurrentHashMap<String, IpPortBasedClient> snapshot = new ConcurrentHashMap<>(newData.size());
             for (Map.Entry<String, ClientSyncData> entry : newData.entrySet()) {
                 IpPortBasedClient snapshotClient = new IpPortBasedClient(entry.getKey(), false);
```
Comment (pixystone) [inline naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:295]: Consider using factory ?

**Review by pixystone — CHANGES_REQUESTED** [review]

File: naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java (hunk 3 above)
Comment (KomachiSion) [reply naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:295]: Using factory will add multiple dependencies. So I think no neccessary.

File: naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java (hunk 1 above)
Comment (KomachiSion) [reply naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:174]: To reduce invalid object create. When the node handle the register persistent instance, the client has been create and register. There is no need to create and try create again.

**Review by horizonzy — APPROVED** [review]

**Pull request merged**